test/integration_test.go

```diff
@@ -512,6 +518,31 @@ func (ts *IntegrationTestSuite) TestDomainUpdate() {
 	ts.Equal(description, *domain.DomainInfo.Description)
 }
 
+func (ts *IntegrationTestSuite) TestNonDeterministicWorkflowFailPolicy() {
+	err := ts.executeWorkflow("test-nondeterminism-failpolicy", ts.workflows.NonDeterminismSimulatorWorkflow, nil)
+	customErr, ok := err.(*internal.CustomError)
+	ts.Truef(ok, "expected CustomError but got %T", err)
+	ts.Equal("NonDeterministicWorkflowPolicyFailWorkflow", customErr.Reason())
```

Comment:
`errors.As` and/or there might be an equivalent in the assertions.  these semantics shouldn't change if/when we add wrapping later.

Reply:
makes sense